packages/pigeon/lib/generator.dart

```diff
@@ -3,11 +3,123 @@
 // found in the LICENSE file.
 
 import 'ast.dart';
+import 'generator_tools.dart';
 
 /// A superclass of generator classes.
 ///
 /// This provides the structure that is common across generators for different languages.
 abstract class Generator<T> {
-  /// Generates files for specified language with specified [languageOptions]
-  void generate(T languageOptions, Root root, StringSink sink);
+  /// Generates files for specified language with specified [generatorOptions]
+  void generate(T generatorOptions, Root root, StringSink sink);
+}
+
+/// A Superclass that enforces code generation across platforms.
+abstract class StructuredGenerator<T> extends Generator<T> {
+  @override
+  void generate(
+    T generatorOptions,
+    Root root,
+    StringSink sink,
+  ) {
+    final Indent indent = Indent(sink);
+
+    prePrologue(generatorOptions, root, sink, indent);
```

Comment:
I was initially thinking it could be for some preprocessing, I ended up putting the code I expected would be here in the `preEnums` method though. So I think this is removable for certain.

Reply:
changed